docs/source/markdown/podman-run.1.md

```diff
@@ -901,20 +901,20 @@ Ulimit options. You can use **host** to copy the current configuration from the
 
 Sets the username or UID used and optionally the groupname or GID for the specified command.
 
-Without this argument the command will be run as root in the container.
+Without this argument, the command will run as the user specified in the container image. Unless overridden by a `USER` command in the Containerfile or by a value passed to this option, this user generally defaults to root. The actual UID of the process running in the container is based on the User namespace the container is running in. By default, if you are running a rootful Podman, the container will run using the UID of the host's root. If you are running the container as a rootless user, by default the UID used by root inside of the container is the rootless user's UID.
```

Comment:
I don't know if this paragraph makes much sense. We shouldn't be focusing on the user the container is run as, but that root in the container is or is not root on the host.

Reply:
This whole paragraph started from a question ni the BZ:  "Namely should lack of `--user` option result in container running as specified by `USER` in Dockerfile? This is what I observe.'

I started the first bit of the USER, then Dansuggested the bit about the UID too and I tweaked it slightly.  I'm happy to make it anything that folks see fit.  Would it work if I just remove the last bit talking about the UID?

```
Without this argument, the command will run as the user specified in the container image. Unless overridden by a `USER` command in the Containerfile or by a value passed to this option, this user generally defaults to root. 
```

Reply:
This sounds like two separate issues.

We need to beef up the documentation of `--user` to detail what happens there; and we need to improve the documentation here to specify what happens when user namespaces are involved. Intermingling the two here seems nonsensical from a reader's perspective, especially given that user namespaces do not just alter the mapping of the user the container is run as, but all users in the container.

Reply:
That is what I've attempted to do here.  Apparently I've missed the mark.  Please give me suggested changes as I'm not sure what to change at this point.

Reply:
I think we need to separate the bit about user/Dockerfile USER and put it in the description of the `--user` flag. This section should focus entirely on how a userns will result in users inside the container having different IDs from users outside the container, how this depends on the exact UID mappings of the user namespace, and how user namespaces are default for rootless and optional for root. Will write more tomorrow.

Reply:
@mheon, this is currently under the --user option, not the --userns option.  Did you get a chance to tickle the keyboard with your thoughts?

Reply:
Having this bit under `--user` and the rest under `--userns` does satisfy my request.

Still, the documentation here seems overly focused on root. Should probably be more generic - "When a user namespace is not in use, the UID and GID used within the container and on the host will match. When user namespaces are in use, however, the UID and GID in the container may correspond to another UID and GID on the host. In rootless containers, for example, a user namespace is always used, and root in the container will by default correspond to the UID and GID of the container invoking Podman."

Reply:
Matt, I've removed this:  
```
When a user namespace is not in use, the UID and GID used within the container and on the host will match. 
When user namespaces are in use, however, the UID and GID in the container may correspond to another 
UID and GID on the host. In rootless containers, for example, a user namespace is always used, and root in
the container will by default correspond to the UID and GID of the container invoking Podman.
```
From the original paragraph.  I then took what you have above and made it a new second paragraph under user, BUT, I changed the second from last word from `container` to `user`

PTAL